### PR TITLE
Cr266 add endpoint to request sms fulfilment

### PR DESCRIPTION
**Pull request by PeterBochel**
# Motivation and Context

# What has changed
Addition of new endpoint to CaseEndpoint, and corresponding business level code in CaseServiceImpl.
Unit tests added.

# How to test?
Run unit tests.

To manually test:

1. Make sure that you GCP project has Firestore enabled. This should be in 'Native' mode.
2. Load a case. See the RH service readme.md for some notes on how to do this. Make sure that the addressType is set to 'HI'.
3. Confirm that case has been loaded into Firestore.
4. Invoke the new endpoint. On the command line this can be done with HTTPie:
```
$ http POST "http://localhost:8071/cases/dc4477d1-dd3f-4c69-b181-7ff725dc9fa4/fulfilments/sms" caseId=dc4477d1-dd3f-4c69-b181-7ff725dc9fa4 telNo=07714111111 fulfilmentCode=UACIT1 dateTime="2016-11-09T11:44:44.797Z"
```
5. Check the fulfilment request message on a queue bound to 'event.fulfilment.request'.

# Links
See: https://collaborate2.ons.gov.uk/jira/browse/CR-266

### CODE REVIEW

**Comment by cookel2** [comment]
The code changes all look good and I have tested that they work correctly in my own GCP environment, which they do. However, I was not able to test any SMS fulfilment requests of addressType=HH (aka caseType of HouseHold) due to these not existing in the product reference (as discussed with Ameet). This is something that I need to speak to @philwhiles about when he is back from leave NB. the problem seems to be that "RH" is missing from the list of requestChannels.
I did successfully test that I could manually publish a CASE_CREATED event, of addressType HI, to the events exchange and see it appear in my GCP Firestore. I then posted a request to the given URL and saw a FULFILMENT_REQUESTED event then appear in the queue associated with the event.fulfilment.request routing key.

**Comment by PeterBochel** [comment]
Readme updated as suggested.

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java [hunk 1]
```diff
@@ -43,4 +60,90 @@
...
+        .with("caseType", caseType)
+        .with("deliveryChannel", deliveryChannel)
+        .with("fulfilmentCode", fulfilmentCode)
+        .debug("Attempting to find product.");
+
+    // Build search criteria base on the cases details and the requested fulfilmentCode
+    Product searchCriteria = new Product();
+    searchCriteria.setRequestChannels(Arrays.asList(Product.RequestChannel.RH));
+    searchCriteria.setRegions(Arrays.asList(region));
+    searchCriteria.setCaseType(caseType);
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java]: Don't use case type in product search, as that will limit products to only HH, and we want to support HI subsequently

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java [hunk 2]
```diff
@@ -43,4 +60,90 @@
...
+  private FulfilmentRequest createFulfilmentRequestPayload(
+      Product product, String telephoneNumber, CollectionCase caseDetails) {
+    // Create the event payload request
+    FulfilmentRequest fulfilmentRequest = new FulfilmentRequest();
+    fulfilmentRequest.setCaseId(caseDetails.getId());
+    if (product.getCaseType().equals(Product.CaseType.HI)) {
+      fulfilmentRequest.setIndividualCaseId(UUID.randomUUID().toString());
+    }
+
+    fulfilmentRequest.setFulfilmentCode(product.getFulfilmentCode());
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java]: Don't set Address and _only_ set tel no in contact

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java (hunk 1 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java]: I've removed case type from the product search.

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java (hunk 2 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java]: ok.
Code is no longer setting address or generic contact details. Only setting phone number in contact details.

**Pull request merged**